server/data/data-jpa/src/main/java/org/apache/james/sieve/jpa/JPASieveRepository.java

```diff
@@ -0,0 +1,337 @@
+                entityManager.persist(jpaSieveScript);
+            } catch (QuotaExceededException | StorageException e) {
+                rollbackTransactionIfActive(entityManager.getTransaction());
+                throw e;
+            }
+        }).sneakyThrow(), throwStorageException("Unable to put script for user " + user.asString()));
+    }
+
+    @Override
+    public List<ScriptSummary> listScripts(User user) throws StorageException {
+        return findAllSieveScriptsForUser(user).stream()
+                .map(JPASieveScript::toSummary)
+                .collect(ImmutableList.toImmutableList());
+    }
+
+    private List<JPASieveScript> findAllSieveScriptsForUser(User user) throws StorageException {
+        return transactionRunner.runAndRetrieveResult(entityManager -> {
+            List<JPASieveScript> sieveScripts = entityManager.createNamedQuery("findAllByUsername", JPASieveScript.class)
+                    .setParameter("username", user.asString()).getResultList();
+            return Optional.ofNullable(sieveScripts).orElse(ImmutableList.of());
+        }, throwStorageException("Unable to list scripts for user " + user.asString()));
+    }
+
+    @Override
+    public ZonedDateTime getActivationDateForActiveScript(User user) throws StorageException, ScriptNotFoundException {
+        Optional<JPASieveScript> script = findActiveSieveScript(user);
+        JPASieveScript activeSieveScript = script.orElseThrow(() -> new ScriptNotFoundException("Unable to find active script for user " + user.asString()));
+        return activeSieveScript.getActivationDateTime().toZonedDateTime();
+    }
+
+    @Override
+    public InputStream getActive(User user) throws ScriptNotFoundException, StorageException {
+        Optional<JPASieveScript> script = findActiveSieveScript(user);
+        JPASieveScript activeSieveScript = script.orElseThrow(() -> new ScriptNotFoundException("Unable to find active script for user " + user.asString()));
+        return IOUtils.toInputStream(activeSieveScript.getScriptContent(), StandardCharsets.UTF_8);
+    }
+
+    private Optional<JPASieveScript> findActiveSieveScript(User user) throws StorageException {
+        return transactionRunner.runAndRetrieveResult(
+                Throwing.<EntityManager, Optional<JPASieveScript>>function(entityManager -> findActiveSieveScript(user, entityManager)).sneakyThrow(),
+                throwStorageException("Unable to find active script for user " + user.asString()));
+    }
+
+    private Optional<JPASieveScript> findActiveSieveScript(User user, EntityManager entityManager) throws StorageException {
+        try {
+            JPASieveScript activeSieveScript = entityManager.createNamedQuery("findActiveByUsername", JPASieveScript.class)
+                    .setParameter("username", user.asString()).getSingleResult();
+            return Optional.ofNullable(activeSieveScript);
+        } catch (NoResultException e) {
```

Comment:
We use to add a log when an exception arise.
For this specific case, I think that it can be a normal case, so you can use the _debug_ level.
`LOGGER.debug("Sieve script not found for user {}", user.asString());`